### PR TITLE
chore(cart): Make all cart entities soft-deletable

### PR DESCRIPTION
**Pull request by olivermrbl**

### CODE REVIEW

**Review by riqwan — APPROVED** [review]
just in time, lgtm!

**Comment by adrien2p** [comment]
Dont we need cascade on soft delete? Maybe not but just wanted to check

**Comment by olivermrbl** [comment]
> Dont we need cascade on soft delete? Maybe not but just wanted to check

I followed the approach taken in the fulfillment module. Did I miss something? 🤔

**Comment by adrien2p** [comment]
> > Dont we need cascade on soft delete? Maybe not but just wanted to check
> 
> 
> 
> I followed the approach taken in the fulfillment module. Did I miss something? 🤔 

There is a task on linear about cascade etc for the fulfilment, it is not finished yet on that part until that ticket is being resolved. But if you want the soft deletion to cascade on the relations it needs to be specified

**Comment by olivermrbl** [comment]
> There is a task on linear about cascade etc for the fulfilment, it is not finished yet on that part until that ticket is being resolved. But if you want the soft deletion to cascade on the relations it needs to be specified

Ah sorry. I wasn't aware of this.

File: packages/cart/src/models/adjustment-line.ts [hunk 1]
```diff
@@ -16,12 +16,6 @@ export default abstract class AdjustmentLine {
   @Property({ columnType: "text", nullable: true })
   description: string | null = null
 
-  @Property({
-    columnType: "text",
-    nullable: true,
-  })
-  promotion_id: string | null = null
-
```
Comment (adrien2p) [inline packages/cart/src/models/adjustment-line.ts:24]: is that intended? feels a bit out of scope no?

File: packages/cart/src/models/line-item-tax-line.ts [hunk 2]
```diff
@@ -11,17 +11,32 @@ import LineItem from "./line-item"
...
   @ManyToOne({
     entity: () => LineItem,
-    index: "IDX_tax_line_item_id",
-    cascade: [Cascade.REMOVE, Cascade.PERSIST]
+    persist: false,
   })
   item: LineItem
 
-  @Property({ columnType: "text" })
+  @Property({ columnType: "text", index: "IDX_tax_line_item_id" })
```
Comment (adrien2p) [inline packages/cart/src/models/line-item-tax-line.ts]: suggestion: deletes at is null

File: packages/cart/src/models/line-item-tax-line.ts [hunk 3]
```diff
@@ -11,17 +11,32 @@ import LineItem from "./line-item"
...
   item_id: string
 
+  @Property({
+    columnType: "text",
+    nullable: true,
+    index: "IDX_line_item_tax_line_tax_rate_id"
```
Comment (adrien2p) [inline packages/cart/src/models/line-item-tax-line.ts]: suggestion: deleted at is null and tax rate id is not null

File: packages/cart/src/models/line-item.ts [hunk 4]
```diff
@@ -23,23 +29,22 @@ type OptionalLineItemProps =
...
 @Entity({ tableName: "cart_line_item" })
+@Filter(DALUtils.mikroOrmSoftDeletableFilterOptions)
 export default class LineItem {
   [OptionalProps]?: OptionalLineItemProps
 
   @PrimaryKey({ columnType: "text" })
   id: string
 
-  @Property({ columnType: "text" })
+  @Property({ columnType: "text", index: "IDX_line_item_cart_id" })
```
Comment (adrien2p) [inline packages/cart/src/models/line-item.ts]: suggestion: deleted at is null maybe to review in all new indexes?

Comment (adrien2p) [inline packages/cart/src/models/line-item.ts]: todo: since the relation is persist false, it will remove the foreign key constraint from the db, maybe that we don't really care but otherwise you need to apply the many to one on the foregin key prop as well, we can discuss if needed

**Review by adrien2p — APPROVED** [review]
LGTM, couple of suggestions to look at 💪

File: packages/cart/src/models/adjustment-line.ts (hunk 1 above)
Comment (olivermrbl) [reply packages/cart/src/models/adjustment-line.ts:24]: It is intended, as we need separate indexes for the inheriting classes

File: packages/cart/src/models/line-item.ts (hunk 4 above)
Comment (olivermrbl) [reply packages/cart/src/models/line-item.ts]: We prob. don't want this persist flag here actually. Will revert.

**Review by sradevski — APPROVED** [review]
